pkg/operator/detectinputchange/as_configmap.go

```diff
@@ -0,0 +1,115 @@
+package detectinputchange
+
+import (
+	"encoding/base64"
+	"fmt"
+	"hash/fnv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+)
+
+func GetConfigMapHash(obj *corev1.ConfigMap) (string, error) {
+	hasher := fnv.New32()
+	_, err := hasher.Write([]byte(WriteConfigMapV1OrDie(obj)))
+	if err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(hasher.Sum(nil)), nil
+}
+
+func GetSecretHash(obj *corev1.Secret) (string, error) {
+	hasher := fnv.New32()
+	_, err := hasher.Write([]byte(WriteSecretV1OrDie(obj)))
+	if err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(hasher.Sum(nil)), nil
+}
+
+func GetStringMapDataHashesForObjects(objs ...runtime.Object) (map[string]string, error) {
+	ret := map[string]string{}
+
+	for _, obj := range objs {
+		switch t := obj.(type) {
+		case *corev1.ConfigMap:
+			hash, err := GetConfigMapHash(t)
+			if err != nil {
+				return nil, err
+			}
+			// this string coercion is lossy, but it should be fairly controlled and must be an allowed name
+			ret[fmt.Sprintf("%s.%s.%s", t.Namespace, t.Name, "configmap")] = hash
```

Comment:
nit: move `fmt.Sprintf("%s.%s.%s", t.Namespace, t.Name, "configmap")` to function